`launch_testing` regressions for Resolute

### Description
On Ubuntu Resolute `launch_testing` is failing with errors

```
E   AssertionError: 'terminating_proc.py-3' not found in "Found multiple processes matching name 'terminating_proc.py' and cmd_args 'None'. Procs: terminating_proc.py-19 /home/jenkins-agent/workspace/ci_linux/ws/install/launch_testing/lib/launch_testing/terminating_proc.py, terminating_proc.py-20 /home/jenkins-agent/workspace/ci_linux/ws/install/launch_testing/lib/launch_testing/terminating_proc.py --extra, terminating_proc.py-21 /home/jenkins-agent/workspace/ci_linux/ws/install/launch_testing/lib/launch_testing/terminating_proc.py node:=different_name"
```
It's possible that this is due to the new free threading support in CPython: https://docs.python.org/3/howto/free-threading-python.html

### Test regressions
[launch_testing.test.launch_testing.test_io_handler_and_assertions.TestIoHandlerAndAssertions.test_assertInStdout_notices_no_matching_proc](https://ci.ros2.org/job/ci_linux/27840/testReport/junit/launch_testing.test.launch_testing.test_io_handler_and_assertions/TestIoHandlerAndAssertions/test_assertInStdout_notices_no_matching_proc/)
[launch_testing.test.launch_testing.test_io_handler_and_assertions.TestIoHandlerAndAssertions.test_assertInStdout_notices_too_many_matching_procs](https://ci.ros2.org/job/ci_linux/27840/testReport/junit/launch_testing.test.launch_testing.test_io_handler_and_assertions/TestIoHandlerAndAssertions/test_assertInStdout_notices_too_many_matching_procs/)
[launch_testing.test.launch_testing.test_io_handler_and_assertions.TestIoHandlerAndAssertions.test_process_names](https://ci.ros2.org/job/ci_linux/27840/testReport/junit/launch_testing.test.launch_testing.test_io_handler_and_assertions/TestIoHandlerAndAssertions/test_process_names/)
[launch_testing.test.launch_testing.test_io_handler_and_assertions.TestIoHandlerAndAssertions.test_regex_matching](https://ci.ros2.org/job/ci_linux/27840/testReport/junit/launch_testing.test.launch_testing.test_io_handler_and_assertions/TestIoHandlerAndAssertions/test_regex_matching/)

--- a/launch_testing/test/launch_testing/test_io_handler_and_assertions.py
+++ b/launch_testing/test/launch_testing/test_io_handler_and_assertions.py
@@ -126,9 +126,9 @@ class TestIoHandlerAndAssertions(unittest.TestCase):
         self.assertTrue(any(contains_ready))
 
     def test_process_names(self):
-        self.assertIn('terminating_proc.py-1', self.proc_output.process_names())
-        self.assertIn('terminating_proc.py-2', self.proc_output.process_names())
-        self.assertIn('terminating_proc.py-3', self.proc_output.process_names())
+        self.assertIn(self.proc_1.name, self.proc_output.process_names())
+        self.assertIn(self.proc_2.name, self.proc_output.process_names())
+        self.assertIn(self.proc_3.name, self.proc_output.process_names())
 
     def test_processes(self):
         self.assertIn(self.proc_1, self.proc_output.processes())
@@ -143,18 +143,18 @@ class TestIoHandlerAndAssertions(unittest.TestCase):
         print(cm.exception)
 
         # Make sure the assertion method lists the names of the process it does have:
-        self.assertIn('terminating_proc.py-1', str(cm.exception))
-        self.assertIn('terminating_proc.py-2', str(cm.exception))
-        self.assertIn('terminating_proc.py-3', str(cm.exception))
+        self.assertIn(self.proc_1.name, str(cm.exception))
+        self.assertIn(self.proc_2.name, str(cm.exception))
+        self.assertIn(self.proc_3.name, str(cm.exception))
 
     def test_assertInStdout_notices_too_many_matching_procs(self):
         with self.assertRaisesRegex(Exception, 'Found multiple processes') as cm:
             assertInStdout(self.proc_output, self.EXPECTED_TEXT, 'terminating_proc.py')
 
         # Make sure the assertion method lists the names of the duplicate procs:
-        self.assertIn('terminating_proc.py-1', str(cm.exception))
-        self.assertIn('terminating_proc.py-2', str(cm.exception))
-        self.assertIn('terminating_proc.py-3', str(cm.exception))
+        self.assertIn(self.proc_1.name, str(cm.exception))
+        self.assertIn(self.proc_2.name, str(cm.exception))
+        self.assertIn(self.proc_3.name, str(cm.exception))
 
     def test_strict_proc_matching_false(self):
         assertInStdout(
@@ -168,7 +168,7 @@ class TestIoHandlerAndAssertions(unittest.TestCase):
         assertInStdout(
             self.proc_output,
             re.compile(r'Called with arguments \S+'),
-            'terminating_proc.py-2'
+            self.proc_2.name
         )
 
     def test_arguments_disambiguate_processes(self):

--- a/launch_testing/test/launch_testing/test_io_handler_and_assertions.py
+++ b/launch_testing/test/launch_testing/test_io_handler_and_assertions.py
@@ -181,7 +181,7 @@ class TestIoHandlerAndAssertions(unittest.TestCase):
 
     def test_asserts_on_missing_text(self):
         with self.assertRaisesRegex(AssertionError, self.NOT_FOUND_TEXT):
-            assertInStdout(self.proc_output, self.NOT_FOUND_TEXT, 'terminating_proc.py-1')
+            assertInStdout(self.proc_output, self.NOT_FOUND_TEXT, self.proc_1.name)
 
     def test_asserts_on_missing_text_by_proc(self):
         with self.assertRaisesRegex(AssertionError, self.NOT_FOUND_TEXT):

